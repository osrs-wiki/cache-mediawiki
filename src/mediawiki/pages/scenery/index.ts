export { default as sceneryPageBuilder } from "./scenery";
export * from "./scenery.types";
