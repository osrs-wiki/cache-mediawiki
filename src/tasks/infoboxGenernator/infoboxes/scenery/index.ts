export * from "./scenery";
export * from "./scenery.types";
