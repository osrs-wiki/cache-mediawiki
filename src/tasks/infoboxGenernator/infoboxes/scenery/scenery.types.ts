import { MediaWikiDate, MediaWikiFile } from "@osrs-wiki/mediawiki-builder";

export type InfoboxScenery = {
  name: string;
  image: MediaWikiFile;
  release: MediaWikiDate | "";
  update: string;
  removal?: MediaWikiDate;
  removalupdate?: MediaWikiDate;
  members?: boolean;
  quest?: string;
  options?: string[];
  respawn?: number;
  icon?: MediaWikiFile;
  map: string;
  examine: string;
  id: string;
};
