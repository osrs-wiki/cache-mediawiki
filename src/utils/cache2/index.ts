export * from "./LazyPromise";
export * from "./types";
