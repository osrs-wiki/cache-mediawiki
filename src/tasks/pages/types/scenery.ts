import { writePageToFile } from "../pages.utils";

import { sceneryPageBuilder } from "@/mediawiki/pages/scenery";
import { CacheProvider, Obj } from "@/utils/cache2";

export const writeSceneryPageFromCache = async (
  cache: Promise<CacheProvider>,
  id: number
) => {
  try {
    const scenery = await Obj.load(cache, id);

    if (scenery) {
      writeSceneryPage(scenery);
    }
  } catch (e) {
    console.error(`Error generating page for scenery ${id}: `, e);
  }
};

export const writeSceneryPage = async (scenery: Obj) => {
  const builder = sceneryPageBuilder(scenery);
  writePageToFile(builder, "scenery", scenery.name, scenery.id.toString());
};
