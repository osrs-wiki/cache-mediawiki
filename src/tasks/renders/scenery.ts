import { existsSync } from "fs";
import { copyFile, mkdir } from "fs/promises";

import Context from "@/context";
import { Obj } from "@/utils/cache2";
import { formatFileName } from "@/utils/files";

export const renderScenery = async (scenery: Obj) => {
  if (scenery.name.toLocaleLowerCase() === "null") {
    return;
  }
  try {
    if (
      Context.renders &&
      existsSync(`./data/${Context.renders}/object/${scenery.id}_orient0.png`)
    ) {
      await mkdir(`./out/${Context.renders}/scenery`, { recursive: true });
      await copyFile(
        `./data/${Context.renders}/object/${scenery.id}_orient0.png`,
        formatFileName(`./out/${Context.renders}/scenery/${scenery.name}.png`)
      );
    }
  } catch (e) {
    console.error(e);
  }
};
