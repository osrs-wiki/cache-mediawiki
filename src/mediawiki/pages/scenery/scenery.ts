import {
  InfoboxTemplate,
  MediaWikiBreak,
  MediaWikiBuilder,
  MediaWikiDate,
  MediaWikiFile,
  MediaWikiTemplate,
  MediaWikiText,
} from "@osrs-wiki/mediawiki-builder";

import { InfoboxScenery } from "./scenery.types";

import Context from "@/context";
import { Obj } from "@/utils/cache2";

export const sceneryPageBuilder = (scenery: Obj) => {
  const infoboxscenery = new InfoboxTemplate<InfoboxScenery>("Scenery", {
    name: scenery.name as string,
    image: new MediaWikiFile(`${scenery.name}.png`),
    release: Context.updateDate
      ? new MediaWikiDate(new Date(Context.updateDate))
      : undefined,
    update: Context.update,
    members: true,
    quest: "No",
    options: scenery.actions,
    examine: Context.examines?.scenery
      ? Context.examines.scenery[scenery.id]
      : "",
    map: "No",
    id: `${Context.beta ? "beta" : ""}${scenery.id.toString()}`,
  });

  const builder = new MediaWikiBuilder();
  builder.addContents([
    new MediaWikiTemplate("New Content"),
    infoboxscenery.build(),
    new MediaWikiFile(`${scenery.name} detail.png`, {
      horizontalAlignment: "left",
      resizing: { width: 130 },
    }),
    new MediaWikiBreak(),
    new MediaWikiText(scenery.name, { bold: true }),
  ]);

  return builder;
};

export default sceneryPageBuilder;
