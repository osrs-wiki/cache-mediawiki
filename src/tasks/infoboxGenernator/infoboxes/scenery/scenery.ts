import {
  InfoboxTemplate,
  MediaWikiBreak,
  MediaWikiBuilder,
  MediaWikiDate,
  MediaWikiFile,
  MediaWikiTemplate,
} from "@osrs-wiki/mediawiki-builder";
import { mkdir, writeFile } from "fs/promises";

import { InfoboxScenery } from "./scenery.types";
import Context from "../../../../context";
import { Obj, CacheProvider } from "../../../../utils/cache2";

const sceneryInfoboxGenerator = async (
  cache: Promise<CacheProvider>,
  id: number
) => {
  try {
    const scenery = await Obj.load(cache, id);

    buildsceneryInfobox(scenery);
  } catch (e) {
    console.error(`Error generating infobox for scenery ${id}: `, e);
  }
};

export const buildsceneryInfobox = async (scenery: Obj) => {
  try {
    const infoboxscenery = new InfoboxTemplate<InfoboxScenery>("Scenery", {
      name: scenery.name as string,
      image: new MediaWikiFile(`${scenery.name}.png`),
      release: Context.updateDate
        ? new MediaWikiDate(new Date(Context.updateDate))
        : undefined,
      update: Context.update,
      members: true,
      quest: "No",
      options: scenery.actions,
      examine: Context.examines?.scenery
        ? Context.examines.scenery[scenery.id]
        : "",
      map: "No",
      id: scenery.id.toString(),
    });

    const builder = new MediaWikiBuilder();
    builder.addContents([
      new MediaWikiTemplate("New Content"),
      infoboxscenery.build(),
      new MediaWikiFile(`${scenery.name} detail.png`, {
        horizontalAlignment: "left",
        resizing: { width: 130 },
      }),
      new MediaWikiBreak(),
    ]);

    await mkdir("./out/infobox/scenery", { recursive: true });
    writeFile(`./out/infobox/scenery/${scenery.id}.txt`, builder.build());
    return builder;
  } catch (e) {
    console.error("Error building scenery infobox: ", e);
  }
};

export default sceneryInfoboxGenerator;
