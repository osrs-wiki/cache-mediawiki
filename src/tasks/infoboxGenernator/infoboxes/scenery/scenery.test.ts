import { buildsceneryInfobox } from "./scenery";
import Context from "../../../../context";
import { ObjID } from "../../../../utils/cache2";

describe("Scenery Infobox", () => {
  it("buildsceneryInfobox", async () => {
    Context.update = "update";
    Context.updateDate = "01-01-1999";
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore Do not require all fields
    const builder = await buildsceneryInfobox({
      name: "name",
      actions: ["action1", "action2"],
      id: 1 as ObjID,
    });
    expect(builder?.build()).toMatchSnapshot();
  });
});
