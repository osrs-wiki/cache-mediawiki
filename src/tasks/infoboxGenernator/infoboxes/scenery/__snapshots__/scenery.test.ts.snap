// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenery Infobox buildsceneryInfobox 1`] = `
"{{New Content}}
{{Infobox Scenery
|name = name
|image = [[File:name.png]]
|release = [[1 January]] [[1999]]
|update = update
|members = Yes
|quest = No
|options = action1, action2
|examine = 
|map = No
|id = 1
}}
[[File:name detail.png|130px|left]]
'''name'''"
`;
